generic/tests/dd_test.py

```diff
@@ -60,62 +69,106 @@ def _get_file(filename, select, test=test):
     error_context.context("Wait guest boot up", logging.info)
     session = vm.wait_for_login(timeout=timeout)
 
-    dd_if = params.get("dd_if")
-    dd_if_select = int(params.get("dd_if_select", '-1'))
-    dd_of = params.get("dd_of")
-    dd_of_select = int(params.get("dd_of_select", '-1'))
-    dd_bs = params.get("dd_bs")
-    dd_count = params.get("dd_count")
-    dd_iflag = params.get("dd_iflag")
-    dd_oflag = params.get("dd_oflag")
-    dd_skip = params.get("dd_skip")
-    dd_seek = params.get("dd_seek")
+    dd_keys = ['dd_if', 'dd_of', 'dd_bs', 'dd_count', 'dd_iflag',
+               'dd_oflag', 'dd_skip', 'dd_seek']
 
-    dd_timeout = int(params.get("dd_timeout", 60))
+    dd_params = {key: params.get(key, None) for key in dd_keys}
+    if dd_params['dd_bs'] is None:
+        dd_params['dd_bs'] = '512'
+    dd_params['dd_bs'] = dd_params['dd_bs'].split()
+    bs_count = len(dd_params['dd_bs'])
 
+    dd_timeout = int(params.get("dd_timeout", 60))
     dd_output = params.get("dd_output", "")
     dd_stat = int(params.get("dd_stat", 0))
 
-    dd_cmd = "dd"
-    if dd_if:
-        dd_if = _get_file(dd_if, dd_if_select)
-        dd_cmd += " if=%s" % dd_if
-    if dd_of:
-        dd_of = _get_file(dd_of, dd_of_select)
-        dd_cmd += " of=%s" % dd_of
-    if dd_bs:
-        dd_cmd += " bs=%s" % dd_bs
-    if dd_count:
-        dd_cmd += " count=%s" % dd_count
-    if dd_iflag:
-        dd_cmd += " iflag=%s" % dd_iflag
-    if dd_oflag:
-        dd_cmd += " oflag=%s" % dd_oflag
-    if dd_skip:
-        dd_cmd += " skip=%s" % dd_skip
-    if dd_seek:
-        dd_cmd += " seek=%s" % dd_seek
-    logging.info("Using '%s' cmd", dd_cmd)
-
-    error_context.context("Execute dd in guest", logging.info)
+    dev_partitioned = []
+    for arg in ['dd_if', 'dd_of']:
+        filename = dd_params[arg]
+        path = _get_file(filename,
+                         int(params.get('%s_select' % arg, '-1')))
+        if (bs_count > 1
+                and filename in params.objects('images')):
+            psize = float(
+                utils_numeric.normalize_data_size(
+                    params.get("partition_size", '2G')
+                    )
+                )
+            start = 0.0
+            dev_id = os.path.split(path)[-1]
+            dev_partitioned.append(dev_id)
+            try:
+                utils_disk.create_partition_table_linux(session, dev_id, 'gpt')
+                for i in range(bs_count):
+                    utils_disk.create_partition_linux(session, dev_id,
+                                                      '%fM' % psize,
+                                                      '%fM' % start)
+                    start += psize
+            except Exception as e:
```

Comment:
This `try..except..` clause is redundant, as it try to catch all the exceptions.
You will get test error without it.